> @sanctum32
> 
> ```
> D:\a\server\server\src\shared\Utilities\Util.h(137,63): error C2039: 'ptr_fun': is not a member of 'std'
> ```

this pr is not related to this error, seems C++17 was enabled in SD3 and that triggered chain errors

--- a/src/game/Server/WorldSocket.cpp
+++ b/src/game/Server/WorldSocket.cpp
@@ -772,7 +772,7 @@ int WorldSocket::HandleAuthSession(WorldPacket& recvPacket)
     sha.UpdateBigNumbers(&K, nullptr);
     sha.Finalize();
 
-    if (sha.GetDigest() != digest)
+    if (memcmp(sha.GetDigest(), digest, SHA_DIGEST_LENGTH))
     {
         WorldPacket packet(SMSG_AUTH_RESPONSE, 1);
         packet << uint8(AUTH_FAILED);

